### DISCUSSION TITLE
Stronger Encryption

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#1387] [Ideas]
We know bitwarden supports PDKF2 in all of its forms.

Might it be possible for use a 128 length or 256 length salt for security purposes. While 64 is sufficient and I can always agree to that, there are some of us that do like enhanced security.

I know PDKF2 also support HMAC-SHA512 which we know is more efficient on modern processors than HMAC-SHA256.

I do have a suggestion. That we can some how make the crypto.rs file adjustable via the docker-compose yaml? Maybe by setting environment variables to control the encryption. However I havent dug into the code sufficiently to see how feasible this is.

I am always open to criticism and discussion on my idea.

**Comment by BlackDex** [comment]
What is your end goal? Better encryption of all the vault passwords?

If that is the case, we can't do anything about that. It is all done client side and we use the web-vault as is (with some minor ux changes).

Also, if we did go and change the web-vault, all the other clients would need support for this as well.

I would suggest to report this upstream, but i think there already is a request line this.
